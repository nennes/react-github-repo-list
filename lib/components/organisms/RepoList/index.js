/* @flow */

import React from "react";
import { Repo as RepoType } from "../../../types";
import Repo from "../../molecules/Repo";

type Props = {
  repoList: Array<RepoType>
};

const RepoList = (props: Props) => {
  return (
    <div className="repo-list">
      <ul>
        {props.repoList.map((repo, index) => {
          return (
            <li key={index}>
              <Repo {...repo} />
            </li>
          );
        })}
      </ul>
    </div>
  );
};

export default RepoList;
