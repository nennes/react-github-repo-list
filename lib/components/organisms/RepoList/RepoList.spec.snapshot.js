/* global test, expect */

import React from "react";
import RepoList from "./index";
import renderer from "react-test-renderer";

test("RepoList organism renders correctly", () => {
  const props = {
    repoList: [
      {
        markdown: "# Some markdown",
        name: "repository-name",
        timestamp: "2015-12-13T13:36:22Z",
        title: "Repository title",
        url: "http://www.test.com"
      },
      {
        markdown: "# Some more markdown",
        name: "another-repository-name",
        timestamp: "2015-12-15T15:33:22Z",
        title: "Another repository title",
        url: "http://www.example.com"
      }
    ]
  };

  const component = renderer.create(<RepoList {...props} />);
  expect(component.toJSON()).toMatchSnapshot();
});
