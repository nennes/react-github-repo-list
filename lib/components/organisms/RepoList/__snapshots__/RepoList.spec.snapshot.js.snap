// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoList organism renders correctly 1`] = `
<div
  className="repo-list"
>
  <ul>
    <li>
      <div
        className="repo"
      >
        <p>
          <div
            className="repo__name"
          >
            <a
              href="http://www.test.com"
              target="_blank"
            >
              <span
                title="repository-name"
              >
                repository-name
              </span>
            </a>
          </div>
        </p>
        <p>
          <span
            className="repo__title"
          >
            <h3>
              Repository title
            </h3>
          </span>
        </p>
        <p>
          <div
            className="repo__read-me"
          >
            <button
              onClick={[Function]}
            >
              ReadMe
            </button>
          </div>
          <div
            className="repo__last-updated"
          >
            Last Updated: 
            <time
              dateTime="2015-12-13T13:36:22.000Z"
              title="2015-12-13T13:36:22Z"
            >
              2 years ago
            </time>
          </div>
        </p>
      </div>
    </li>
    <li>
      <div
        className="repo"
      >
        <p>
          <div
            className="repo__name"
          >
            <a
              href="http://www.example.com"
              target="_blank"
            >
              <span
                title="another-repository-name"
              >
                another-repository-name
              </span>
            </a>
          </div>
        </p>
        <p>
          <span
            className="repo__title"
          >
            <h3>
              Another repository title
            </h3>
          </span>
        </p>
        <p>
          <div
            className="repo__read-me"
          >
            <button
              onClick={[Function]}
            >
              ReadMe
            </button>
          </div>
          <div
            className="repo__last-updated"
          >
            Last Updated: 
            <time
              dateTime="2015-12-15T15:33:22.000Z"
              title="2015-12-15T15:33:22Z"
            >
              2 years ago
            </time>
          </div>
        </p>
      </div>
    </li>
  </ul>
</div>
`;
